web/react/components/code_preview.jsx

```diff
@@ -0,0 +1,276 @@
+        }
+
+        this.setState({code: '', lang: usedLanguage, canLoad: true});
+
+        $.ajax({
+            async: true,
+            url: props.fileUrl,
+            type: 'GET',
+            error: this.handleReceivedError,
+            success: this.handleReceivedCode
+        });
+    }
+
+    handleReceivedCode(data) {
+        this.setState({code: data});
+    }
+
+    handleReceivedError() {
+        this.setState({canLoad: false});
+    }
+
+    static formatCode(lang, data, filename) {
+        var language = lang || '';
+        var parsed;
+        var header = '';
+
+        language = language.toLowerCase();
+
+        if (HighlightedLanguages[language]) {
+            var name = HighlightedLanguages[language].name;
+
+            if (filename) {
+                const fname = decodeURIComponent(Utils.getFileName(filename));
+                name = fname + ' - ' + name;
+            }
+
+            header = '<span class="post-body--code__language">' + name + '</span>';
+
+            try {
+                parsed = hlJS.highlight(language, data).value;
+            } catch (e) {
+                parsed = TextFormatting.sanitizeHtml(data);
+            }
+        } else {
+            parsed = TextFormatting.sanitizeHtml(data);
+        }
+
+        const lines = data.match(/\r\n|\r|\n|$/g).length;
+        var strlines = '';
+        for (var i = 1; i <= lines; i++) {
+            if (strlines) {
+                strlines += '\n' + i;
+            } else {
+                strlines += i;
+            }
+        }
+
+        return (
+            '<div class="post-body--code">' +
+                header +
+                '<pre>' +
+                    '<code class="hljs">' +
+                        '<table>' +
+                            '<tbody>' +
+                                '<tr>' +
+                                    '<td class="post-body--code__lineno">' + strlines + '</td>' +
+                                    '<td>' +
+                                        parsed +
+                                    '</td>' +
+                                '</tr>' +
+                            '</tbody>' +
+                        '</table>' +
+                    '</code>' +
+                '</pre>' +
+            '</div>'
+        );
+    }
+
+    static support(filename) {
+        return typeof getlang(filename) !== 'undefined';
+    }
+
+    render() {
+        if (!this.state.lang || !this.state.canLoad) {
+            return (
+                <FileInfoPreview
+                    filename={this.props.filename}
+                    fileUrl={this.props.fileUrl}
+                    fileInfo={this.props.fileInfo}
+                    formatMessage={this.props.formatMessage}
+                />
+            );
+        }
+
+        return <div dangerouslySetInnerHTML={{__html: CodePreview.formatCode(this.state.lang, this.state.code, this.props.filename)}}/>;
+    }
+}
+
+function getlang(filename) {
```

Comment:
Can you rename this to getLang so we can keep it consistent with everything else being in camelCase?